Sources/eppo/ApiEndpoints.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+import os
+
+/// Handles API endpoint URL construction with subdomain support.
+public class ApiEndpoints {
+    private static let logger: Logger = {
+        let bundleID = Bundle.main.bundleIdentifier ?? "com.eppo.sdk"
+        return Logger(subsystem: bundleID, category: String(describing: ApiEndpoints.self))
+    }()
+    
+    private let effectiveBaseURL: String
+    
+    public init(baseURL: String?, sdkToken: String) {
+        let tokenDecoder = SdkTokenDecoder(sdkToken)
+        if baseURL == defaultHost {
+            Self.logger.warning("[Eppo SDK] custom baseURL cannot be the default host")
```

Comment:
```suggestion
            Self.logger.warning("[Eppo SDK] custom baseURL cannot be the default host")
```

This seems actionable. Do you feel value in telling the user what actual host is going to be used so they are not surprised?

Reply:
Hm. Maybe it's not worth it to warn here at all since I don't want to surface the effective domain (i.e. subdomain insertion) here in the code. The use of the baseUrl param just needs to be well explained in the docs and during onboarding for those proxying and we won't have any surprises on either side.